src/Integration/LocalServices/IUnboundSolutionChecker.cs

```diff
@@ -31,25 +35,42 @@ internal interface IUnboundSolutionChecker
         /// <summary>
         /// Returns true/false if the currently opened solution/folder is bound and requires re-binding
         /// </summary>
-        bool IsBindingUpdateRequired();
+        Task<bool> IsBindingUpdateRequired(CancellationToken token);
     }
 
     internal class UnboundSolutionChecker : IUnboundSolutionChecker
     {
         private readonly IExclusionSettingsStorage exclusionSettingsStorage;
+        private readonly IConfigurationProvider bindingConfigProvider;
+        private readonly ISonarQubeService sonarQubeService;
         private readonly ILogger logger;
 
-        public UnboundSolutionChecker(IExclusionSettingsStorage exclusionSettingsStorage, ILogger logger)
+        public UnboundSolutionChecker(IExclusionSettingsStorage exclusionSettingsStorage,
+            IConfigurationProvider bindingConfigProvider,
+            ISonarQubeService sonarQubeService,
+            ILogger logger)
         {
             this.exclusionSettingsStorage = exclusionSettingsStorage;
+            this.bindingConfigProvider = bindingConfigProvider;
+            this.sonarQubeService = sonarQubeService;
             this.logger = logger;
         }
 
-        public bool IsBindingUpdateRequired()
+        public async Task<bool> IsBindingUpdateRequired(CancellationToken token)
         {
             try
             {
-                return !exclusionSettingsStorage.SettingsExist();
+                var savedExclusions = exclusionSettingsStorage.GetSettings();
+
+                if (savedExclusions == null)
+                {
+                    return true;
+                }
+
+                var bindingConfiguration = bindingConfigProvider.GetConfiguration();
+                var serverExclusions = await sonarQubeService.GetServerExclusions(bindingConfiguration.Project.ProjectKey, token);
```

Comment:
If we are in standalone mode, then GetSettings would return null and we will never get here. So Project must be not null here.

Reply:
I think it's worth adding an assertion to document this assumption.